File: fixtures/scale/scala/large/src/cryptobench/ciphers/xtea/XteaSuite.scala
package cryptobench.ciphers.xtea

import cryptobench.core.Cipher
import cryptobench.core.CipherSuite
import cryptobench.core.SuiteResult
import cryptobench.verify.RoundTrip

final class XteaSuite extends CipherSuite {
  override def name: String = "xtea"

  override def run(): SuiteResult = {
    val cipher: Cipher = new XteaCipher(XteaKey.default())
    var passed = 0
    var failed = 0
    val start = System.nanoTime()
    for (sample <- XteaVectors.samples()) {
      if (RoundTrip.check(cipher, sample)) passed += 1 else failed += 1
    }
    SuiteResult(name, passed, failed, System.nanoTime() - start)
  }
}

File: fixtures/scale/dart/large/lib/ciphers/xtea/xtea_suite.dart
import '../../core/cipher.dart';
import '../../core/cipher_suite.dart';
import '../../core/suite_result.dart';
import '../../verify/round_trip.dart';
import 'xtea_cipher.dart';
import 'xtea_key.dart';
import 'xtea_vectors.dart';

class XteaSuite implements CipherSuite {
  @override
  String name() => 'xtea';

  @override
  String category() => 'cipher';

  @override
  SuiteResult run() {
    final Cipher cipher = XteaCipher(XteaKey.defaultKey());
    var passed = 0;
    var failed = 0;
    final start = DateTime.now();
    for (final sample in xteaSamples()) {
      if (roundTripCheck(cipher, sample)) {
        passed++;
      } else {
        failed++;
      }
    }
    final elapsed = DateTime.now().difference(start).inMicroseconds;
    return SuiteResult(name(), passed, failed, elapsed);
  }
}
